integreat_compass/cms/views/index.py

```diff
@@ -1,9 +1,93 @@
-from django.views.generic import TemplateView
+from functools import reduce
 
+from django.shortcuts import render
+from django.views.generic import ListView
 
-class IndexView(TemplateView):
+from ..constants import offer_group_types, offer_mode_types
+from ..models import Offer, Tag
+
+
+class IndexListView(ListView):
     """
-    Start page of the project
+    View to see all offers
     """
 
     template_name = "index.html"
+    context_object_name = "all_offers_list"
+
+    def get_queryset(self):
+        """
+        :return: The offer queryset
+        :rtype: ~django.db.models.query.QuerySet
+        """
+        return Offer.objects.all()
+
+    def get_context_data(self, **kwargs):
+        """
+        This functions displays all offers
+        :param kwargs: The supplied keyword arguments
+        :type kwargs: dict
+
+        :return: The context of all
+        :rtype: dict
+        """
+        offers = Offer.objects.all()
+        tags = Tag.objects.all()
+        print(type(Offer.objects.all()))
+        context = {
+            "tags": tags,
+            "offers": offers,
+            "offer_group_types": offer_group_types.CHOICES,
+            "offer_mode_types": offer_mode_types.CHOICES,
+        }
+        return context
+
+    def get(self, request, *args, **kwargs):
+        """
+        This functions filters
+
+        :param request: The current request
+        :type request: ~django.http.HttpRequest
+
+        :param args: The supplied arguments
+        :type args: list
+
+        :param kwargs: The supplied keyword arguments
+        :type kwargs: dict
+
+        :return: Response for filtered offers
+        :rtype: ~django.template.response.TemplateResponse
+        """
+        tags = Tag.objects.all()
+        filtered_offers = Offer.objects.all()
```

Comment:
```suggestion
        filtered_offers = Offer.objects..exclude(public_version__isnull=True)
```
Something along these lines is necessary so we do not attempt to show new/non-public offers